job-service-config/src/main/java/com/hpe/caf/services/configuration/AppConfig.java

```diff
@@ -51,10 +66,14 @@ public String getDatabaseUsername(){
         return environment.getProperty("JOB_SERVICE_DATABASE_USERNAME");
     }
 
-    public String getDatabasePassword(){
-        return environment.getProperty("JOB_SERVICE_DATABASE_PASSWORD");
+    public String getDatabasePassword() {
+        try {
+            return SECRETS_CACHE.get("JOB_SERVICE_DATABASE_PASSWORD");
+        } catch (final ExecutionException e) {
+            throw new RuntimeException("Failed to get secret for 'JOB_SERVICE_DATABASE_PASSWORD'", e);
```

Comment:
Would throwing an `AppConfigException` be better here?

```suggestion
            throw new AppConfigException("Failed to get secret for 'JOB_SERVICE_DATABASE_PASSWORD'", e);
```